### PR TITLE
[17.0][MIG] product_harmonized_system

### PR DESCRIPTION
**Pull request by AndreuOForgeFlow**
Standard migration

@ForgeFlow

### CODE REVIEW

File: product_harmonized_system/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# Copyright 2022 Tecnativa - Víctor Martínez
+# @author Benjamin Henquet <info@noviat.com>
+# @author Kumar Aberer <kumar.aberer@braintec-group.com>
+# @author Alexis de Lattre <alexis.delattre@akretion.com>
+# @author Luc de Meyer <info@noviat.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Product Harmonized System Codes",
+    "version": "17.0.1.1.0",
```
Comment (GuillemCForgeFlow) [inline product_harmonized_system/__manifest__.py]:
```suggestion
    "version": "17.0.1.0.0",
```

File: product_harmonized_system/views/product_template.xml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        <field name="arch" type="xml">
+            <group name="group_general" position="inside">
+                <field name="hs_code_id" invisible="type == 'service'" />
+                <field
+                    name="origin_country_id"
+                    invisible="type == 'service'"
+                    options="{'no_create': True}"
+                />
+            </group>
+        </field>
```
Comment (GuillemCForgeFlow) [inline product_harmonized_system/views/product_template.xml:22]:
```suggestion
        <field name="inherit_id" ref="intrastat_base.product_template_form_view" />
        <field name="arch" type="xml">
            <field name="is_accessory_cost" position="after">
                <field name="hs_code_id" invisible="type == 'service'" />
                <field
                    name="origin_country_id"
                    invisible="type == 'service'"
                    options="{'no_create': True}"
                />
            </field>
        </field>
```
I don't see a reason why the H.S. Code and Country of Origin shouldn't be on the Intrastat section of the Product Template form view, declared in the `intrastat_base` module: https://github.com/OCA/intrastat-extrastat/blob/17.0/intrastat_base/views/product_template.xml#L8C17-L8C43.
If done, we could update the readme files to let the people know about it.
Any comments about this? @alexis-via @luc-demeyer

File: product_harmonized_system/views/product_template.xml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    <record id="product_template_search_view" model="ir.ui.view">
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_search_view" />
+        <field name="arch" type="xml">
+            <filter name="categ_id" position="after">
+                <filter
+                    string="H.S. Code"
+                    name="hs_code_groupby"
+                    context="{'group_by': 'hs_code_id'}"
+                />
```
Comment (GuillemCForgeFlow) [inline product_harmonized_system/views/product_template.xml:34]:
```suggestion
                <filter
                    string="H.S. Code"
                    name="hs_code_groupby"
                    context="{'group_by': 'hs_code_id'}"
                />
                <filter
                    string="Country of Origin"
                    name="origin_country_groupby"
                    context="{'group_by': 'origin_country_id'}"
                />
```
I guess we can also add the Country of Origin group by, just as an extra improvement.

**Review by GuillemCForgeFlow — COMMENTED** [review]
the rest is fine, code review 👍🏽

**Review by GuillemCForgeFlow — APPROVED** [review]
LGTM! Code and functional review 👍🏽

**Review by JasminSForgeFlow — APPROVED** [review]
Code Review

**Comment by GuillemCForgeFlow** [comment]
@pedrobaeza perhaps this is ready to be merged? 😄 
So we can start with the migration for the rest of the modules, such as `product_harmonized_system_delivery`, `product_harmonized_system_delivery` and `hs_code_link`.

**Review by pedrobaeza — APPROVED** [review]
Let's go!

/ocabot merge nobump

**Comment by alexis-via** [comment]
@GuillemCForgeFlow hs_code_link is a duplicate of product_harmonized_system_stock_delivery (rename of product_harmonized_system_delivery for v17). So please don't port hs_code_link !

File: product_harmonized_system/security/ir.model.access.csv [hunk 4]
```diff
@@ -0,0 +1,3 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_hs_code_group_system,Full access on hs.code to Settings group,model_hs_code,base.group_system,1,1,1,1
+access_hs_code_read,Read access on hs.code to everybody,model_hs_code,,1,0,0,0
```
Comment (alexis-via) [inline product_harmonized_system/security/ir.model.access.csv]: in v17, you have a warning when you have an ACL with an empty group

**Comment by GuillemCForgeFlow** [comment]
@alexis-via you're right, thank you for the notice 👍🏽

**Comment by alexis-via** [comment]
You can take inspiration from my v17 migration PR which was cancelled because it had the 3 modules (product_harmonized_system, product_harmonized_system_stock and product_harmonized_system_stock_delivery) in the same PR : here is the v17 migration commit https://github.com/OCA/intrastat-extrastat/pull/258/commits/806c27bdef58a01c2537ac8aa38e324fff445555  That's how I noticed that the update of ir.model.access was missing.

**Comment by GuillemCForgeFlow** [comment]
> You can take inspiration from my v17 migration PR which was cancelled because it had the 3 modules (product_harmonized_system, product_harmonized_system_stock and product_harmonized_system_stock_delivery) in the same PR : here is the v17 migration commit [806c27b](https://github.com/OCA/intrastat-extrastat/commit/806c27bdef58a01c2537ac8aa38e324fff445555) That's how I noticed that the update of ir.model.access was missing.

Great, that helps a lot! 😄 thank you for the heads up.

**Comment by alexis-via** [comment]
@GuillemCForgeFlow I thought you were aware of my previous v17 migration work on those modules... I should have mentionned it earlier, it would have saved you some work time!

**Comment by pedrobaeza** [comment]
@AndreuOForgeFlow your last changes are not merged into the main branch, as you did the push after launching the command. If they are interesting, please do another IMP PR.

**Comment by AndreuOForgeFlow** [comment]
> @AndreuOForgeFlow your last changes are not merged into the main branch, as you did the push after launching the command. If they are interesting, please do another IMP PR.

Oh okay! Will do

**Pull request closed without merging**